slurp.py

```diff
@@ -942,18 +959,21 @@ def matches( rule, kwargs={} ):
                 base2 = base1.split('_')[-2:]
                 input_datasets[ '_'.join(base2) ] = 1
 
-    
+   
     if len(lfn_lists)==0: return [], None, []  # Early exit if nothing to be done
     
+
     # Build dictionary of DSTs existing in the datasets table of the file catalog.  For every DST that is in this list,
     # we know that we do not have to produce it if it appears w/in the outputs list.
     dsttype="%s_%s_%s"%(name,build,tag)  # dsttype aka name above
     
     exists = {}
     INFO("Building list of existing outputs")
-    chkq = f"select filename,runnumber,segment from datasets where runnumber>={runMin} and runnumber<={runMax} and filename like'"+dsttype+"%';"
-    for check in dbQuery( cnxn_string_map['fccro'], chkq ):
-        exists[ check.filename ] = ( check.runnumber, check.segment)  # key=filename, value=(run,seg)
+
+    for output_, tuple_ in dstnames.items():
+        dt, ds = tuple_
+        exists = { c.filename : ( c.runnumber, c.segment ) for c in fccro.execute( f"select filename, runnumber, segment from datasets where runnumber>={runMin} and runnumber<={runMax} and dsttype='{dt}' and dataset='{ds}'" ) }
```

Comment:
This is very pythonic... but we should be using the standard dbQuery function for the database access.